src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -248,10 +248,18 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
                 //renew retention lease with global checkpoint so that any shard that picks up shard replication task has data until then.
                 try {
                     retentionLeaseHelper.renewRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint, followerShardId)
+                    lastLeaseRenewalMillis = System.currentTimeMillis()
```

Comment:
> It is better to colocate the last renewal time in the method itself so we don't run into out of sync or forgotten paths if we introduce newer renewal path.

RetentionLeaseHelper in itself doesn't really hold any state for each shard. We can keep renewal time per shard but the current approach seems simpler IMHO.   
Also, It is not about renewal but rather more around if we can ignore the error for some time and fail only if the issue persists for more than  configured duration.

Reply:
> Can you elaborate why this was the issue? Can the globalCheckpoint being lower than local seqNo happen in renewal scenarios as well?


We tried debugging this issue. We don't know why exactly the global checkpoint was lower in this particular issue. We're also not able to reproduce the issue. 
We also had a discussion offline  and based on the discussion, we decided to make these changes to renew the lease(instead of adding) and also handling the `RetentionLeaseInvalidRetainingSeqNoException` during renewal.


During renewal, typically, we won't have this issue where retention lease is higher than global checkpoint. But we've added it in renewal path as renewal can fail for the first time due to the current issue as well.